java/util/HashMap/HashMap.hpp

```diff
@@ -127,11 +164,12 @@ namespace Java {
 
 			/**
 			 * Removes the mapping for the specified key & value from this map if present.
-			 * @param key
-			 * @param value
-			 * @return
+			 *
+			 * @param K key
+			 * @param V value
+			 * @return boolean
 			 */
-			boolean remove(K key, const V value) {
+			boolean remove(K key, V value) {
 				if (NULL == get(key) || !containsValue(value)) {

```

Comment:
Removes the entry for the specified key **only if it is currently mapped to the specified value.**
https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html#remove-java.lang.Object-java.lang.Object-
Please make sure it works as expected and write test.